apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreQuery.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"):
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using static Google.Cloud.Datastore.V1.ReadOptions.Types;
+
+namespace Google.Cloud.Datastore.V1;
+
+/// <summary>
+/// A Datastore query with additional options for features such as read consistency and profiling.
+/// </summary>
+public sealed class DatastoreQuery
```

Comment:
Should this be inmutable? I know we have the ToRequest methods but still...

Reply:
I wondered about this - but the underlying protos aren't, so it would give an incorrect impression IMO.
I did consider writing something like "This is mutable, but any instances provided to the library are never mutated, so you can reuse them if you don't mutate them yourself" but figured it would be too weird.

Reply:
Yep, that's true about the protos.